analytics/js_tracking_api.rst

```diff
@@ -0,0 +1,842 @@
+
+Example of usage::
+
+    _paq.push(["addEcommerceItem", "craft-311", "Unicorn Iron on Patch", "Crafts & Sewing", "499", "3"]);
+
+Tracking Ecommerce order
+````````````````````````
+To track successful ecommerce order (on checkout page for example) use ``trackEcommerceOrder`` function::
+
+    _paq.push(["trackEcommerceOrder", orderId, orderGrandTotal, orderSubTotal, orderTax, orderShipping, orderDiscount]);
+
+.. data:: orderId
+
+    **Required** Unique order ID.
+
+.. data:: orderGrandTotal
+
+    **Required** Order Revenue grand total  - tax, shipping and discount included, written as number - not string.
+
+.. data:: orderSubTotal
+
+    **Optional** Order sub total - without shipping, written as number - not string.
+
+.. data:: orderTax
+
+    **Optional** Order tax amount written as number - not string.
+
+.. data:: orderShipping
+
+    **Optional** Order shipping costs, written as number - not string.
+
+.. data:: orderDiscount
+
+    **Optional** Order discount amount, written as number - not string.
+
+Example of usage::
+
+    _paq.push(["trackEcommerceOrder", "3352", 499, 399, 0, 100]);
+
+Updating cart
+`````````````
+To update user cart (when user adds new product or removes them from cart) use ``trackEcommerceCartUpdate`` function::
+
+    _paq.push(["trackEcommerceCartUpdate", cartAmount]);
+
+.. data:: cartAmount
+
+    Cart amount, written as number, required.
+
+
+.. warning::
+
+    Before tracking cart update be sure you added all products in cart by using ``addEcommerceItem`` first.
+    (Even ones that were in cart earlier) Then use this function last.
+
+Example of usage::
+
+        _paq.push(["trackEcommerceCartUpdate", 250]);
+
+Tracking product / category view
+````````````````````````````````
+If you want to track when user enters product site, or is browsing products category use ``setEcommerceView`` function::
+
+    _paq.push(["setEcommerceView", productSKU, productName, productCategory, productPrice]);
+
+.. data:: productSKU
+
+    String with product stock-keeping unit, required parameter. False for tracking category.
+
+.. data:: productName
+
+    **Optional** String with product name. False for tracking category.
+
+.. data:: productCategory
+
+    **Optional** Product category, can be written as Array with up to 5 elements.
+
+.. data:: productPrice
+
+    **Optional** String with product price.
+
+.. warning::
+
+    All string parameters should be escaped, without double quote character.
+
+Example of usage::
+
+    _paq.push(["setEcommerceView", "craft-311", "Unicorn Iron on Patch", "Crafts & Sewing", "499"]);
+
+
+Custom Variables
+^^^^^^^^^^^^^^^^
+
+Adding Custom Variable
+``````````````````````
+To set custom variable that can be used later use ``setCustomVariable`` function::
+
+    _paq.push(["setCustomVariable", index, name, value, scope]);
+
+.. data:: index
+
+    Number from 1 to 5 where variable is stored
+
+.. data:: name
+
+    Name of the variable
```

Comment:
I'm confused. I thought that variable is stored in slot (configured in Piwik before first use), but here we're sending index and name of custom variable. What happens when name doesn't match configured name of variable? Is it overwritten, ignored or does each name have separate key-value inside each index? I don't see it used in other methods for custom variables (delete and read). What's the point of this value?